terraform/consul/README.md

```diff
@@ -0,0 +1,139 @@
+
+What you'll gain from this demo:
+
+- Grasp the strategic advantage of Infrastructure as Code (IaC) through Consul and Terraform, enabling rapid, consistent, and error-free provisioning of network resources which can significantly accelerate the time-to-market.
+- Dive into the automated service deployment and DNS record updates enabled by Consul-Terraform Sync, illustrating a model for maintaining real-time service availability and domain resolution which are essential for business continuity.
+- Practical insights into the seamless integration between DNSimple, Consul, and Terraform, providing a holistic view of the network infrastructure automation process.
+
+## Prerequisites
+
+- A zone on DNSimple (e.g. `example.com`) - whether it's registered with DNSimple or not. It doesn't need to be managed by DNSimple's name servers, but you won't be able to see the changes live. Only using DNSimple's UI. Don't have an account? [Sign up free](https://dnsimple.com/sign_up).
+  - If you'd like to register and set up a domain entirely through Terraform, check out our [domains demo](../domains).
+- API access token with write access to the zone. Create a new token by [following our support article](https://support.dnsimple.com/articles/api-access-token/). Note the token and your account ID, as you will need them later.
+- Docker and Docker Compose installed on your machine. [Install Docker](https://docs.docker.com/get-docker/).
+
+## Getting Started
+
+### Step 1: Prepare the configuration files
+
+Copy the `terraform.tfvars.example` file to `terraform.tfvars` and fill in the required variables.
+
+```bash
+cp config/terraform.tfvars.example config/terraform.tfvars
+cp config/api-service.json.example config/api-service.json
+cp config/web-service.json.example config/web-service.json
+```
+
+Next, you would need to update the `api-service.json` and `web-service.json` files with the zone that you have chosen to use for this demo. You can also take this chance to inspect the configuration of the services and familiarize yourself with the syntax and options available.
+
+### Step 2: Create the infrastructure
+
+In this demo, we will be using Docker Compose to create the infrastructure. This will create a Consul server and two Consul clients. The Consul clients will be running the `api-service` and `web-service` containers. In addition, we will also be running the consul-terraform-sync container which will be responsible for syncing the Consul service catalog with the Terraform state.
```

Comment:
```suggestion
In this demo, we will be using Docker Compose to create the infrastructure. This will create a Consul server and two Consul clients. The Consul clients will be running the `api-service` and `web-service` containers. We will also be running the consul-terraform-sync container, which will be responsible for syncing the Consul service catalog with the Terraform state.
```